src/ontodia/diagram/view.ts

```diff
@@ -432,9 +432,122 @@ export class DiagramView extends Backbone.Model {
                 end: this.createLinkMarker(linkTypeId, false, style.markerTarget),
             };
         }
+        style.router = this.getLinkRouter();
         return style;
     }
 
+    private getLinkRouter(): LinkRouter {
+        return (vertices: Vertex[], args: RouterProps, linkView: LinkView): Vertex[] => {
+            const link = linkView.model;
+            if (!vertices) {
+                vertices = [];
+            }
+            // If the cell is a view, find its model.
+
+            // The cell is a link. Let's find its source and target models.
+            let srcId = link.get('source').id || link.previous('source').id;
+            let trgId = link.get('target').id || link.previous('target').id;
+
+            // Use the same direction for all siblings
+            const direction = srcId > trgId;
+            if (direction) {
+                const temp = srcId;
+                srcId = trgId;
+                trgId = temp;
+            }
+
+            // If one of the ends is not a model, the link has no siblings.
+            if (!srcId || !trgId) {
+                return vertices;
+            }
+
+            const links: Link[] = getLinksBetweenElements (this.model, srcId, trgId);
+            if (links.length < 2) {
+                return vertices;
+            }
+
+            const siblings: Link[] = links.filter((l) => {
+                const verts = l.get('vertices');
+                if (verts && verts.length !== 0) {
+                    return false;
+                }
+                const _srcId = l.get('source').id;
+                const _trgId = l.get('target').id;
+
+                return (_srcId === srcId && _trgId === trgId) ||
+                       (_srcId === trgId && _trgId === srcId);
+            });
+
+            // There is more than one siblings. We need to create vertices.
+            // First of all we'll find the middle point of the link.
+            const srcCenter = this.model.graph.getCell(srcId).getBBox().center();
+            const trgCenter = this.model.graph.getCell(trgId).getBBox().center();
+            const midPoint = joint.g.line(srcCenter, trgCenter).midpoint();
+
+            // Then find the angle it forms.
+            const theta = srcCenter.theta(trgCenter);
+
+            // This is the maximum distance between links
+            const gap = 20;
+            // For mor beautifull positioning
+            const indexModifyer = siblings.length % 2 ? 0 : 1;
+
+            const myIndex = siblings.indexOf(link);
+            if (myIndex !== -1) {
+                const v = updateSibling(link, myIndex);
+
+                vertices = [v];
+            }
+
+            let i = 0;
+            for (const sib of siblings) {
```

Comment:
Use `siblings.forEach((sibling, index) => ...)` instead of `for-of` with `let i`.